### PR TITLE
fix: Correctly handle list values in _python_value_to_proto_value

### PR DESCRIPTION
**Pull request by TBourtonBumble**
Currently, `_python_value_to_proto_value` does not properly handle lists. It will always return a list of values of length one, regardless of the number of input values.

This PR aims to fix this bug, and also adds a check to ensure that the number of returned proto values is the same as the inputted values.

### CODE REVIEW

File: sdk/python/feast/type_map.py [hunk 1]
```diff
@@ -372,11 +372,15 @@ def _python_value_to_proto_value(
                 if feast_value_type == ValueType.BYTES_LIST:
                     raise _type_err(sample, ValueType.BYTES_LIST)
 
-                json_value = json.loads(sample)
-                if isinstance(json_value, list):
+                json_sample = json.loads(sample)
```
Comment (franciscojavierarceo) [inline sdk/python/feast/type_map.py:375]: Any chance you can add a test here?

Comment (TBourtonBumble) [reply sdk/python/feast/type_map.py:375]: Hi, sure thing. Did you have in mind something to check that `json.loads(sample)` does not error out? E.g. if there's something like a json decode error?

**Review by franciscojavierarceo — APPROVED** [review]

File: sdk/python/feast/type_map.py (hunk 1 above)
Comment (franciscojavierarceo) [reply sdk/python/feast/type_map.py:375]: Actually, it's fine. There's no test there currently so no need to add additional work here. Thank you!

**Comment by TBourtonBumble** [comment]
Hi sorry, I missed something very dumb with the unit test, i've pushed a fix so they should be ok to re-run

**Pull request merged**